src/main/ApplicationImpl.cpp

```diff
@@ -728,6 +731,12 @@ ApplicationImpl::getStatusManager()
     return *mStatusManager;
 }
 
+LedgerStateRoot&
+ApplicationImpl::getLedgerStateRoot()
+{
+    return *mLedgerStateRoot;
```

Comment:
I would suggest asserting that this is only used in the main thread otherwise it will potentially corrupt things (code is not thread safe)